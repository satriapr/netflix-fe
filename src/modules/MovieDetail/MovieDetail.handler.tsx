import { useEffect, useState } from 'react'

import Constant from '../../constants/Constant'
import { IMovieDetailRes } from '../../models'
import movieService from '../../services/movie.service'

const initialObj = {
  Title: Constant.NA,
  Year: Constant.NA,
  Rated: Constant.NA,
  Released: Constant.NA,
  Runtime: Constant.NA,
  Genre: Constant.NA,
  Director: Constant.NA,
  Writer: Constant.NA,
  Actors: Constant.NA,
  Plot: Constant.NA,
  Language: Constant.NA,
  Country: Constant.NA,
  Awards: Constant.NA,
  Poster: Constant.NA,
  Ratings: [
    {
      Source: Constant.NA,
      Value: Constant.NA
    }
  ],
  Metascore: Constant.NA,
  imdbRating: Constant.NA,
  imdbVotes: Constant.NA,
  imdbID: Constant.NA,
  Type: Constant.NA,
  totalSeasons: Constant.NA,
  Response: Constant.NA,
  BoxOffice: Constant.NA
}

// Hooks
const Handler = (id: string) => {
  const [movie, setMovie] = useState<IMovieDetailRes>(initialObj)
  const [isLoading, setIsLoading] = useState<boolean>(true)

  useEffect(() => {
    const getMovie = async () => {
      const response = await movieService.getMovie(id)
      setMovie(response)
      setIsLoading(false)
    }

    getMovie()
  }, [id])

  return { isLoading, movie, setMovie }
}

export default Handler
