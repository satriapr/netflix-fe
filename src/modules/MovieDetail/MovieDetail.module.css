.backHeight {
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
}

.content {
  width: 50%;
  margin: 0 auto;
  padding: 20px;
  min-height: 410px;
  font-size: 18px;
  color: #e6e6e6;
  font-weight: 500;
}

.content h1 {
  text-align: center;
  margin: 30px;
  margin-top: 10px;
}

.content p {
  line-height: 19px;
}

.movieDetailImage {
  text-align: center
}

.runAndRate {
  text-align: center;
  font-size: 14px;
  padding-right: 6px;
}

.run {
  padding-right: 6px;
}

.rate {
  border-left: 1px solid white;
  padding-left: 6px;
}

.overviewContainer {
  margin-top: 10px;
}

.greyed {
  color: #9c9c9c;
}

@media only screen and (max-width: 900px) {
  .backHeight {
    height: 100vh;
  }

  /* movie detail */
  .content {
    width: 90%;
    font-size: 93%;
  }

  .content h1 {
    margin: 30px 0;
  }
}

@media only screen and (max-height: 800px) {
  .backHeight {
    height: 100%;
  }
}

@media only screen and (max-width: 350px) {
  .backHeight {
    height: 100%;
  }
}