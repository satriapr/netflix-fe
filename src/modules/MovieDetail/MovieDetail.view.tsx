import { faStar } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import React from 'react'

import Constant from './../../constants/Constant'
import movieDetailHandler from './MovieDetail.handler'
import styles from './MovieDetail.module.css'

const MovieDetail = props => {
  const { id } = props.match.params
  const handler = movieDetailHandler(id)
  const { isLoading } = handler
  const {
    Title,
    Released,
    Runtime,
    Genre,
    Director,
    imdbRating,
    Actors,
    Plot,
    Poster,
    BoxOffice
  } = handler.movie

  const content = (
    <div className={styles.backHeight}>
      <div className={styles.content}>
        <h1>{Title}</h1>

        <div className={styles.movieDetailImage}>
          <img src={Poster} alt={Title} />
        </div>

        <p className={styles.runAndRate}>
          <span className={styles.run}> {Runtime}</span>
          <span className={styles.rate}>
            <FontAwesomeIcon icon={faStar} /> {imdbRating}
          </span>
        </p>
        <div className={styles.overviewContainer}>
          <p>{Plot}</p>

          <p>
            <span className={styles.greyed}>{Constant.RELEASE_DATE}: </span>
            {Released}
          </p>

          <p>
            <span className={styles.greyed}>{Constant.GENRE}: </span>
            {Genre}
          </p>

          <p>
            <span className={styles.greyed}>{Constant.DIRECTOR}: </span>{' '}
            {Director}
          </p>

          <p>
            <span className={styles.greyed}>{Constant.ACTORS}: </span>
            {Actors}
          </p>

          <p>
            <span className={styles.greyed}>{Constant.TOTAL_REVENUE}: </span>
            {BoxOffice}
          </p>
        </div>
      </div>
    </div>
  )

  return (
    <React.Fragment>
      {!isLoading ? content : <div className="loading-mask" />}
    </React.Fragment>
  )
}

export default MovieDetail
