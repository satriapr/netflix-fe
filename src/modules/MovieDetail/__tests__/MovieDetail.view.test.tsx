import MovieDetail from '../MovieDetail.view'
import { testSnapshots } from '../../../utils/test.util'

describe('MovieDetail Component', () => {
  testSnapshots(MovieDetail, [
    {
      description: 'MovieDetail props',
      props: {
        match: {
          params: {
            id: '123'
          }
        }
      }
    }
  ])
})
