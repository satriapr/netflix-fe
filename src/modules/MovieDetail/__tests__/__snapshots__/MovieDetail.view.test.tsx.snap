// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Component snapshots MovieDetail props 1`] = `
<Fragment>
  <div
    className="loading-mask"
  />
</Fragment>
`;
